tests/env/components/dns_resolver.go

```diff
@@ -33,28 +33,38 @@ func (h *handler) ServeDNS(w dns.ResponseWriter, r *dns.Msg) {
 	msg := dns.Msg{}
 	msg.SetReply(r)
 	glog.Infof("dns query:\n%+v", r)
-	switch r.Question[0].Qtype {
-	case dns.TypeA:
-		msg.Authoritative = true
-		domain := msg.Question[0].Name
-
-		address, ok := h.records[domain]
-		if ok {
-			msg.Answer = append(msg.Answer, &dns.A{
-				Hdr: dns.RR_Header{Name: domain, Rrtype: dns.TypeA, Class: dns.ClassINET, Ttl: 60},
-				A:   net.ParseIP(address),
-			})
-			glog.Infof("dns answer: %+v", msg.Answer)
-		} else {
-			msg.Rcode = dns.RcodeNameError
-			glog.Infof("dns return code: %v", dns.RcodeToString[msg.Rcode])
+	msg.Authoritative = true
+	domain := msg.Question[0].Name
+	addresses, ok := h.records[domain]
+	if ok {
+		for _, address := range addresses {
+			ip := net.ParseIP(address)
+			if ip.To4() == nil {
+				// address is IPv6
+				if r.Question[0].Qtype != dns.TypeA {
```

Comment:
I don't fully understand, what is the difference between this section of code and the ipv4 section? Why do you need to check the Qtype?

Reply:
I see. Why not flip the nested if to check `r.Question[0].Qtype == TypeAAAA`? It will be easier to understand.

Maybe you can clarify this in the code comments for future reader

Reply:
There's also TypeANY, in this case IPv6 should be included too.

Reply:
Code changed and comment added for clarity.